models/src/node/crd/mod.rs

```diff
@@ -0,0 +1,100 @@
+pub mod v1;
+pub mod v2;
+
+pub use self::v2::{
+    BottlerocketShadow, BottlerocketShadowSpec, BottlerocketShadowState, BottlerocketShadowStatus,
+};
+
+use crate::node::{error, error::Error};
+
+use k8s_openapi::apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition;
+use kube::CustomResourceExt;
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use std::fmt;
+use std::fmt::Debug;
+
+lazy_static! {
+    static ref BOTTLEROCKETSHADOW_CRD_METHODS: Vec<fn() -> CustomResourceDefinition> = {
+        // A list of CRD methods for different BottlerocketShadow version
+        // The latest version should be added at the end of the vector.
+        let mut crd_methods = Vec::new();
+        crd_methods.push(v1::BottlerocketShadow::crd as fn()->CustomResourceDefinition);
+        crd_methods.push(v2::BottlerocketShadow::crd as fn()->CustomResourceDefinition);
+        crd_methods
+    };
+}
+
+pub trait Selector {
+    fn selector(&self) -> error::Result<BottlerocketShadowSelector>;
+}
+
+impl<T: kube::ResourceExt> Selector for T {
```

Comment:
I think blanket implementing this for all `ResourceExt` is too broad, since owner references could be any type of kubernetes object, and this function will probably not do what people expect if they aren't using it on *our* resources specifically.

Should we require our BottlerocketShadow CRDs to all extend an additional trait, which requires `ResourceExt` be implemented?

e.g.

```rust
trait BottlerocketShadowResource: kube::ResourceExt {}
```

Even if the trait doesn't have any methods, it could be used as a [marker trait](doc.rust-lang.org/std/marker/index.html) to denote what types are implemented by this `Selector` trait.